staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types_discovery.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// APIGroupDiscoveryList is a resource containing a list of APIGroupDiscoveries.
+// The shape is chosen to be very similar to "normal" api endpoints and work well with our type system
+// This is what is returned from the /discovery/v2 endpoint.
+type APIGroupDiscoveryList struct {
+	TypeMeta `json:",inline"`
+	// ResourceVersion will not be set, because this does not have a replayable ordering among multiple apiservers.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
```

Comment:
we don't support protobuf for discovery documents, should we have the protobuf tags?

Reply:
why not support protobuf?

Reply:
> we don't support protobuf for discovery documents, should we have the protobuf tags?

looks like we do...

```
curl -s -H "Accept: application/vnd.kubernetes.protobuf" http://localhost:8001/api/v1 --output - | tail -c +5 | protoc --decode_raw
1 {
  1: ""
  2: "APIResourceList"
}
2 {
  1: "v1"
  2 {
    1: "bindings"
    2: 1
    3: "Binding"
    4 {
      1: "create"
    }
    6: ""
    8: ""
    9: ""
    10: ""
  }
  2 {
    1: "componentstatuses"
    2: 0
    3: "ComponentStatus"
    4 {
      1: "get"
      1: "list"
    }
    5: "cs"
    6: ""
    8: ""
    9: ""
    10: ""
  }
  2 {
    1: "configmaps"
    2: 1
    3: "ConfigMap"
    4 {
      1: "create"
      1: "delete"
      1: "deletecollection"
      1: "get"
      1: "list"
      1: "patch"
      1: "update"
      1: "watch"
    }
    5: "cm"
    6: ""
    8: ""
    9: ""
    10: "qFsyl6wFWjQ="
  }
...
```

Reply:
Cool!